openapi-contracts/src/main/com/sinch/sdk/domains/conversation/models/dto/v1/AgentDto.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Conversation API | Sinch
+ * Send and receive messages globally over SMS, RCS, WhatsApp, Viber Business, Facebook messenger and other popular channels using the Sinch Conversation API.  The Conversation API endpoint uses built-in transcoding to give you the power of conversation across all supported channels and, if required, full control over channel specific features.
+ *
+ * The version of the OpenAPI document: 1.0
+ * Contact: support@sinch.com
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+package com.sinch.sdk.domains.conversation.models.dto.v1;
+
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import java.util.Objects;
+
+/** Represents an agent that is involved in a conversation. */
+@JsonPropertyOrder({
+  AgentDto.JSON_PROPERTY_DISPLAY_NAME,
+  AgentDto.JSON_PROPERTY_TYPE,
+  AgentDto.JSON_PROPERTY_PICTURE_URL
+})
+@JsonFilter("uninitializedFilter")
+@JsonInclude(value = JsonInclude.Include.CUSTOM)
+public class AgentDto {
+  public static final String JSON_PROPERTY_DISPLAY_NAME = "display_name";
+  private String displayName;
+  private boolean displayNameDefined = false;
+
+  public static final String JSON_PROPERTY_TYPE = "type";
+  private String type;
+  private boolean typeDefined = false;
+
+  public static final String JSON_PROPERTY_PICTURE_URL = "picture_url";
+  private String pictureUrl;
+  private boolean pictureUrlDefined = false;
+
+  public AgentDto() {}
+
+  public AgentDto displayName(String displayName) {
+    this.displayName = displayName;
+    this.displayNameDefined = true;
+    return this;
+  }
+
+  /**
+   * Agent&#39;s display name
+   *
+   * @return displayName
+   */
+  @JsonProperty(JSON_PROPERTY_DISPLAY_NAME)
+  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
+  public String getDisplayName() {
+    return displayName;
+  }
+
+  @JsonIgnore
+  public boolean getDisplayNameDefined() {
+    return displayNameDefined;
+  }
+
+  @JsonProperty(JSON_PROPERTY_DISPLAY_NAME)
+  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
+  public void setDisplayName(String displayName) {
+    this.displayName = displayName;
+    this.displayNameDefined = true;
+  }
+
+  public AgentDto type(String type) {
+    this.type = type;
+    this.typeDefined = true;
+    return this;
+  }
+
+  /**
+   * Agent&#39;s classification. It can be UNKNOWN_AGENT_TYPE, HUMAN or BOT.
+   *
+   * @return type
+   */
+  @JsonProperty(JSON_PROPERTY_TYPE)
+  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
+  public String getType() {
```

Comment:
It should be an enum according to the description.
Note: our generator needs to support `x-enumDescriptions` in order to general the correct type

Reply:
Not at this level: At DTO level, all enums are handled as String; this to enable SDK to handle new unsupported values dynamically if server is updated but SDK or depending APP are not yet synchronized

Some kind of 'relax' mode to avoid existing an Application based onto SDK become unusable in production  due to a server release
About `x-enumDescriptions` : the `x-` prefix mean it is a proprietary exception case only handled by who know what this mean. Ensuring having an `enum` could be a way to investigate